### PR TITLE
Improving Clean Architecture application for history use cases

### PR DESCRIPTION
**Pull request by Arnnav-A**
I have added the required state and view model classes for the use cases, along with moving the UI calls to the view class instead of the output data.

### CODE REVIEW

**Review by JG808 — APPROVED** [review]

File: src/interface_adapter/clear/ClearPresenter.java [hunk 1]
```diff
@@ -8,16 +8,21 @@
 public class ClearPresenter implements ClearOutputBoundary {
     private final SearchViewModel searchViewModel;
```
Comment (nicolasdiasm) [inline src/interface_adapter/clear/ClearPresenter.java:9]: Add a line.

File: src/interface_adapter/clear/ClearPresenter.java [hunk 2]
```diff
@@ -8,16 +8,21 @@
...
-    public ClearPresenter(SearchViewModel searchViewModel, ViewManagerModel viewManagerModel) {
+    public ClearPresenter(SearchViewModel searchViewModel, ViewManagerModel viewManagerModel, ClearViewModel clearViewModel) {
         this.searchViewModel = searchViewModel;
         this.viewManagerModel = viewManagerModel;
+        this.clearViewModel = clearViewModel;
     }
 
     @Override
     public void prepareView(ClearOutputData response) {
+        ClearState state =new ClearState();
```
Comment (nicolasdiasm) [inline src/interface_adapter/clear/ClearPresenter.java:21]: Add a space.

File: src/interface_adapter/clear/ClearViewModel.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package interface_adapter.clear;
+
+import interface_adapter.ViewModel;
+
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeSupport;
+
+public class ClearViewModel extends ViewModel {
```
Comment (nicolasdiasm) [inline src/interface_adapter/clear/ClearViewModel.java:8]: Add a new line.

File: src/interface_adapter/clear/ClearViewModel.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private ClearState state = new ClearState();
```
Comment (nicolasdiasm) [inline src/interface_adapter/clear/ClearViewModel.java:9]: Add a new line.

File: src/interface_adapter/clear/ClearViewModel.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    public ClearViewModel() {
+        super("clear");
+    }
```
Comment (nicolasdiasm) [inline src/interface_adapter/clear/ClearViewModel.java:12]: Add a new line.

File: src/interface_adapter/clear/ClearViewModel.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    public void setState(ClearState state) {
+        this.state = state;
+    }
+
+    public ClearState getState() {
+        return this.state;
+    }
+
+    private final PropertyChangeSupport support = new PropertyChangeSupport(this);
```
Comment (nicolasdiasm) [inline src/interface_adapter/clear/ClearViewModel.java:21]: Add a new line.

File: src/interface_adapter/get_history/GetHistoryPresenter.java [hunk 7]
```diff
@@ -7,16 +7,20 @@
 
 public class GetHistoryPresenter implements GetHistoryOutputBoundary {
```
Comment (nicolasdiasm) [inline src/interface_adapter/get_history/GetHistoryPresenter.java:8]: Add a new line.

File: src/interface_adapter/get_history/GetHistoryPresenter.java [hunk 8]
```diff
@@ -7,16 +7,20 @@
...
     private final SearchViewModel searchViewModel;
+    private final GetHistoryViewModel getHistoryViewModel;
     private ViewManagerModel viewManagerModel;
-    public GetHistoryPresenter(SearchViewModel searchViewModel, ViewManagerModel viewManagerModel) {
+    public GetHistoryPresenter(SearchViewModel searchViewModel, ViewManagerModel viewManagerModel, GetHistoryViewModel getHistoryViewModel) {
```
Comment (nicolasdiasm) [inline src/interface_adapter/get_history/GetHistoryPresenter.java:12]: Add a new line before it.

File: src/use_case/clear_history/ClearOutputData.java [hunk 9]
```diff
@@ -1,9 +1,7 @@
 package use_case.clear_history;
 
-import javax.swing.*;
-
 public class ClearOutputData {
```
Comment (nicolasdiasm) [inline src/use_case/clear_history/ClearOutputData.java:3]: Add a new line.

**Review by nicolasdiasm — CHANGES_REQUESTED** [review]
Mainly formatting issues.
Also change the name of the packages of clear to indicate that it is related to search history

**Pull request merged**